crates/holochain/tests/tests/countersigning.rs

```diff
@@ -130,6 +136,186 @@ async fn listen_for_countersigning_completion() {
+            [alice.agent_pubkey().clone()],
+            None,
+            Duration::from_secs(30),
+        )
+        .await
+        .unwrap();
+
+    // Set up the session and accept it for both agents
+    let preflight_request: PreflightRequest = conductors[0]
+        .call_fallible(
+            &alice_zome,
+            "generate_countersigning_preflight_request_fast",
+            vec![
+                (alice.agent_pubkey().clone(), vec![Role(0)]),
+                (bob.agent_pubkey().clone(), vec![]),
+            ],
+        )
+        .await
+        .unwrap();
+    let alice_acceptance: PreflightRequestAcceptance = conductors[0]
+        .call_fallible(
+            &alice_zome,
+            "accept_countersigning_preflight_request",
+            preflight_request.clone(),
+        )
+        .await
+        .unwrap();
+    let alice_response =
+        if let PreflightRequestAcceptance::Accepted(ref response) = alice_acceptance {
+            response
+        } else {
+            unreachable!();
+        };
+    let bob_acceptance: PreflightRequestAcceptance = conductors[1]
+        .call_fallible(
+            &bob_zome,
+            "accept_countersigning_preflight_request",
+            preflight_request.clone(),
+        )
+        .await
+        .unwrap();
+    let bob_response = if let PreflightRequestAcceptance::Accepted(ref response) = bob_acceptance {
+        response
+    } else {
+        unreachable!();
+    };
+
+    // Take Bob's app offline so that Alice can't get his activity
+    conductors[1]
+        .disable_app("app".into(), DisabledAppReason::User)
+        .await
+        .unwrap();
+
+    // Alice shouldn't be able to commit yet, because she doesn't have Bob's activity
+    let result: ConductorApiResult<(ActionHash, EntryHash)> = conductors[0]
+        .call_fallible(
+            &alice_zome,
+            "create_a_countersigned_thing_with_entry_hash",
+            vec![alice_response.clone(), bob_response.clone()],
+        )
+        .await;
+    match result {
+        Ok(_) => {
+            panic!("Expected commit to fail due to missing dependencies");
+        }
+        Err(ConductorApiError::CellError(CellError::WorkflowError(e))) => {
+            match *e {
+                WorkflowError::SourceChainError(SourceChainError::IncompleteCommit(
+                    IncompleteCommitReason::DepMissingFromDht(_),
+                )) => {
+                    // Expected
+                }
+                _ => {
+                    panic!("Expected IncompleteCommit error, got: {:?}", e);
+                }
+            }
+        }
+        _ => {
+            panic!("Expected CellError::WorkflowError, got: {:?}", result);
+        }
+    }
+
+    // Bring Bob's app back online
+    conductors[1].enable_app("app".into()).await.unwrap();
+
+    // Let's make sure Bob will have Alice's activity before committing his countersigning entry
+    let _: AgentActivity = conductors[1]
```

Comment:
Doesn't this always succeed regardless of whether the desired activity could be gotten or not?

Reply:
Yes, if any authority could be contacted then we might get an agent activity response that is not as fresh as we'd like. We'd only get an error if we failed to connect to any authorities.

Because there are only two conductors in this test, and as long as we've done peer discovery, we will always use the other conductor as our only option for an authority and end up getting the latest agent activity. So I'm fairly sure this is sound as a method for syncing up two agents.

It's actually not 100% necessary, because by the same logic, a get request should also always work... So maybe I should slim down the test a bit and lose this?

Reply:
You could, maybe it's more explicit. So the point is to have the acceptance action in your cache? Actually you can just edit the comment to mention what exactly you're getting from the other peer.

Reply:
I've done both, removed the unnecessary `get_agent_activity` calls and updated the comments to better explain what's being fetched.

It's not the acceptance that's being fetched no, it's the chain head for the other peer. Countersigning needs to permit validation to check the agent's state at the point they began countersigning. I guess it's technically the whole chain they might validate but there's only one commit created that we won't have here so it is just the chain head. An app might use this for example, to check that somebody's balance permits the transaction that is being done with countersigning.

Reply:
Got it, good stuff!